pytest/tests/loadtest/locust/common/base.py

```diff
@@ -0,0 +1,327 @@
+        return f"{prefix}_user{id}.{cls.funding_account.key.account_id}"
+
+    def __init__(self, environment):
+        super().__init__(environment)
+        host, port = self.host.split(":")
+        self.node = cluster.RpcNode(host, port)
+        self.id = NearUser.get_next_id()
+        self.account_id = NearUser.account_id(self.id)
+
+    def on_start(self):
+        """
+        Called once per user, creating the account on chain
+        """
+        self.account = Account(key.Key.from_random(self.account_id))
+        self.send_tx_retry(
+            CreateSubAccount(NearUser.funding_account,
+                             self.account.key,
+                             balance=5000.0))
+        self.account.refresh_nonce(self.node)
+
+    def send_tx(self, tx: Transaction):
+        """
+        Send a transaction and return the result, no retry attempted.
+        """
+        block_hash = base58.b58decode(self.node.get_latest_block().hash)
+        signed_tx = tx.sign_and_deser(block_hash)
+
+        # doesn't work because it raises on status etc
+        # rpc_result = self.node.send_tx_and_wait(signed_tx, timeout=DEFAULT_TRANSACTION_TTL_SECONDS)
+
+        params = [base64.b64encode(signed_tx).decode('utf8')]
+        j = {
+            "method": "broadcast_tx_commit",
+            "params": params,
+            "id": "dontcare",
+            "jsonrpc": "2.0"
+        }
+
+        # with self.node.session.post(
+        # This is tracked by locust
+        with self.client.post(
+                url="http://%s:%s" % self.node.rpc_addr(),
+                json=j,
+                timeout=DEFAULT_TRANSACTION_TTL_SECONDS,
+                catch_response=True,
+        ) as response:
+            try:
+                rpc_result = json.loads(response.content)
+                tx_result = evaluate_rpc_result(rpc_result)
+                tx.transaction_id = tx_result["transaction_outcome"]["id"]
+                logger.debug(
+                    f"{tx.transaction_id} for {self.account_id} is successful: {tx_result}"
+                )
+            except NearError as err:
+                logging.warn(f"marking an error {err.message}, {err.details}")
+                response.failure(err.message)
+        return response
+
+    def send_tx_retry(self, tx: Transaction):
+        """
+        Send a transaction and retry until it succeeds
+        """
+        # expected error: UNKNOWN_TRANSACTION means TX has not been executed yet
+        # other errors: probably bugs in the test setup (e.g. invalid signer)
+        # this method is very simple and just retries no matter the kind of
+        # error, as long as it is one defined by us (inherits from NearError)
+        while True:
+            try:
+                result = self.send_tx(tx)
+                return result
+            except NearError:
+                logger.warn(
+                    f"transaction {tx.transaction_id} failed: {result}, retrying in 0.25s"
```

Comment:
I don't think `result` will ever have value other than `None`, so you rather need to catch an exception with

```
except NearError as error:
```
and print it instead